src/main/java/org/zalando/nakadi/controller/EventStreamController.java

```diff
@@ -189,6 +211,7 @@ public StreamingResponseBody streamEvents(
                 writeProblemResponse(response, outputStream, INTERNAL_SERVER_ERROR, e.getMessage());
             } finally {
                 connectionReady.set(false);
+                consumerLimitingService.releaseConnectionSlots(connectionSlots);
```

Comment:
If the feature is disabled then the list `connectionSlots` will be empty and nothing will happen.
But if the feature was enabled when connection started and got disabled before the connection was closed - it's better to release the slots that were acquired.